src/gui/src/layoutViewer.cpp

```diff
@@ -1983,6 +1992,46 @@ void LayoutViewer::drawBlock(QPainter* painter,
   }
 }
 
+void LayoutViewer::drawRegionOutlines(QPainter* painter)
+{
+  if (!options_->areRegionsVisible()) {
+    return;
+  }
+
+  painter->setPen(QPen(Qt::gray, 0));
+  painter->setBrush(Qt::BrushStyle::NoBrush);
+
+  for (auto* region : block_->getRegions()) {
+    const odb::Rect region_box = getRegionRect(region);
+    if (region_box.area() > 0) {
+      painter->drawRect(region_box.xMin(), region_box.yMin(), region_box.dx(), region_box.dy());
+    }
+  }
+}
+
+odb::Rect LayoutViewer::getRegionRect(odb::dbRegion* region) const
+{
+  if (region == nullptr) {
+    return {};
+  }
+
+  auto boundaries = region->getBoundaries();
+  if (boundaries.empty()) {
+    return getRegionRect(region->getParent());
```

Comment:
Will this result in drawing a region twice as you are looping over all regions in drawRegionOutlines?

Reply:
good point, should have been addressed by just drawing the rects of the regions without merging